### ISSUE TITLE
Application crashes at the end of execution when using GPU (OpenVINO 2020.4)

### ISSUE THREAD

**Issue by cavusmustafa** [tensorflow/ngraph-bridge#675]
When the backend device set to GPU, the application crashed just before the execution finishes. The crash happens in the destructor of "IE_Backend" when calling "m_exec_map.clear();" (line 44 of ie_backend.cc).

The issue is observed when using a GPU only. CPU and HDDL runs without any error.

The crash happens at the end of the execution. I was able to execute the inference and see the results.

The backend device is set by the line below in the python script:
ngraph_bridge.set_backend(<device_name>)
 
The error message:
terminate called without an active exception
./run_inception_v3.sh: line 7:  7113 Aborted                 (core dumped) python -u infer_frozen.py -m inception_v3.pb -i input:0 -o InceptionV3/Predictions/Softmax:0 --images images.txt --num-iterations 10 --inf-time -d $device

Models used to observe this issue:
resnet_v2_50
inception_v3

The commit id of the branch I was using: 15a32baf51ec8e435889de45589a32f7878a09d0
OpenVINO version: 2020.4

**Comment by cavusmustafa** [comment]
The issue might be related to "m_exec_map" in IE_Backend. The pointers to the executables are already caches in "m_ng_exec_map" in NGraphEncapsulateImpl. Simply not adding any entry to "m_exec_map" prevents this error.
